src/EventTarget.ts

```diff
@@ -41,28 +41,21 @@ export class GlobalAttributesChangedEvent extends Event {
   }
 }
 
-export interface SplunkOtelWebEventTarget {
-  _experimental_addEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void);
-  _experimental_addEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
-
-  _experimental_removeEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void);
-  _experimental_removeEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
-}
+export interface InternalEventTarget {
+  addEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void): void;
+  addEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void): void;
 
-export class NativeEventTarget implements SplunkOtelWebEventTarget {
-  // we're using DOM API to save on bytes a little bit
-  // using EventTarget directly has low browser compatibility
-  private readonly target = document.createElement('div');
+  removeEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void): void;
+  removeEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void): void;
 
-  _experimental_addEventListener(name: SplunkOtelWebEventType, callback: (event: never) => void): void {
-    this.target.addEventListener(name, callback);
-  }
+  dispatchEvent(event: SessionIdChangedEvent | GlobalAttributesChangedEvent);
+}
 
-  _experimental_removeEventListener(name: SplunkOtelWebEventType, callback: (event: never) => void): void {
-    this.target.removeEventListener(name, callback);
-  }
+export interface SplunkOtelWebEventTarget {
+  _experimental_addEventListener: InternalEventTarget['addEventListener'];
+  _experimental_removeEventListener: InternalEventTarget['removeEventListener']
+}
 
-  dispatchEvent(event: SessionIdChangedEvent | GlobalAttributesChangedEvent): void {
-    this.target.dispatchEvent(event);
-  }
+export function buildInternalEventTarget(): InternalEventTarget {
+  return document.createElement('div');
```

Comment:
Maybe a comment for in 3 months when everyone's forgotten about this

```suggestion
  // EventTarget can't be initiated in older versions of safari and IE so use something that already implements it
  return document.createElement('div');
```